src/webgl/light.js

```diff
@@ -492,4 +492,344 @@ p5.prototype.lightFalloff = function(
   return this;
 };
 
+/**
+ * Creates a spotlight with a given color, position, direction of light,
+ * angle and concentration. Here, angle refers to the opening or aperture
+ * of the cone of the spotlight, and concentration is used to focus the
+ * light towards the center. Both angle and concentration are optional, but if
+ * you want to provide concentration, you will also have to specify the angle.
+ *
+ * @method spotLight
+ * @param  {Number}    v1       red or hue value (depending on the current
+ * color mode),
+ * @param  {Number}    v2       green or saturation value
+ * @param  {Number}    v3       blue or brightness value
+ * @param  {Number}    x        x axis position
+ * @param  {Number}    y        y axis position
+ * @param  {Number}    z        z axis position
+ * @param  {Number}    rx       x axis direction of light
+ * @param  {Number}    ry       y axis direction of light
+ * @param  {Number}    rz       z axis direction of light
+ * @param  {Number}    [angle]  optional parameter for angle. Defaults to PI/3
+ * @param  {Number}    [conc]   optional parameter for concentration. Defaults to 100
+ * @chainable
+ *
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   setAttributes('perPixelLighting', true);
```

Comment:
Part of me feels like we should set 'perPixelLighting' to true by default. These lines can feel unnecessarily complicating for the lighting references.

Reply:
I agree, and I think the per-vertex lighting looks pretty ugly. I also feel like the setAttributes function is a little arcane.

Reply:
Yeah I am not sure if there are performance hits with per pixel. The `setAttributes` function is definitely intended to only be used by people that want a non-traditional webgl setup for whatever reason. The most commonly used attribute modification would theoretically be 'antialias' which would likely be changed by `smooth()` for most users.

I'll break this out into an issue.